openmc/tallies.py

```diff
@@ -259,24 +264,69 @@ def with_summary(self):
 
     @property
     def sum(self):
+        if not self._statepoint:
+            return None
+
+        if not self._results_read:
+            # Extract Tally data from the file
+            data = self._statepoint._f['tallies/tally {0}/results'.format(
+                self.id)].value
+            sum = data['sum']
+            sum_sq = data['sum_sq']
+
+            # Define a routine to convert 0 to 1
+            def nonzero(val):
+                return 1 if not val else val
+
+            # Reshape the results arrays
+            new_shape = (nonzero(self.num_filter_bins),
+                         nonzero(self.num_nuclides),
+                         nonzero(self.num_score_bins))
+
+            sum = np.reshape(sum, new_shape)
+            sum_sq = np.reshape(sum_sq, new_shape)
+
+            # Set the data for this Tally
+            self._sum = sum
+            self._sum_sq = sum_sq
+
+            # Indicate that Tally results have been read
+            self._results_read = True
+
         return self._sum
 
     @property
     def sum_sq(self):
+        if not self._statepoint:
```

Comment:
If you like my recommendation above about just returning the data array, you will need to separately build the ``sum_sq`` array here.